builtin/plugins/coin/coin.go

```diff
@@ -213,6 +257,97 @@ func mint(ctx contract.Context, to loom.Address, amount *loom.BigUInt) error {
 	return ctx.Set(economyKey, econ)
 }
 
+//Mint : to be called by CoinPolicyManager Responsible to mint coins as per various parameter defined
+func Mint(ctx contract.Context) error {
+	div := loom.NewBigUIntFromInt(10)
+	div.Exp(div, loom.NewBigUIntFromInt(18), nil)
+	var policy Policy
+	var amount *common.BigUInt
+	err := ctx.Get(policyKey, &policy)
+	if err != nil {
+		return errUtil.Wrap(err, "Failed to Get PolicyInfo")
+	}
+	changeRatioNumerator := loom.NewBigUIntFromInt(int64(policy.ChangeRatioNumerator))
+	changeRatioDenominator := loom.NewBigUIntFromInt(int64(policy.ChangeRatioDenominator))
+	blockHeight := loom.NewBigUIntFromInt(ctx.Block().Height)
+	baseAmount := loom.NewBigUIntFromInt(int64(policy.TotalSupply))
+	baseAmount.Mul(baseAmount, div)
+	blocksGeneratedPerYear := loom.NewBigUIntFromInt(int64(policy.BlocksGeneratedPerYear))
+	basePercentage := loom.NewBigUIntFromInt(int64(policy.BasePercentage))
+	operator := policy.Operator
+	_, modulus := big.NewInt(ctx.Block().Height).DivMod(big.NewInt(ctx.Block().Height), big.NewInt(int64(policy.BlocksGeneratedPerYear)), big.NewInt(int64(policy.BlocksGeneratedPerYear)))
+	year := blockHeight.Div(blockHeight, blocksGeneratedPerYear)
+	year = year.Add(year, loom.NewBigUIntFromInt(1))
+	var checkMintAmount = &types.BigUInt{
+		Value: *loom.NewBigUIntFromInt(0),
+	}
+	//Checks if minting amount is computed before or it is being computed for first time
+	err = ctx.Get(mintingAmountKey, checkMintAmount)
```

Comment:
Where are you using this `checkMintAmount`?  Also, if you find the amount set in context, you are still going to do entire calculation right?

Reply:
Also, make this calculation more modular, so it's easy to understand.